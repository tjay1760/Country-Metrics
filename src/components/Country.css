/* stylelint-disable indentation */
.country-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  top: 25%;
  position: absolute;
}

.country-box {
  width: 100%;
  height: 100%;
  margin: 0%;
}

  .even {
    background-color: rgb(225, 236, 209);
  }

.flag {
  width: 70%;
}

.search {
  position: absolute;
  width: fit-content;
  margin-top: 5%;
  top: 8%;
}

.country-name {
  text-decoration: none;
}
