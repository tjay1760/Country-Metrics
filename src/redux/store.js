import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './countries/countrySlice';

const store = configureStore({
  reducer: {
    countries: countryReducer,
  },
});

export default store;
