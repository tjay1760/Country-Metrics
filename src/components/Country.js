import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import fetchCountries from '../redux/countries/fetchCountries';
import filterData from './filterData';
import './Country.css';

const Country = () => {
  const dispatch = useDispatch();
  const countriesLoaded = useSelector(
    (state) => state.countries.countries.length > 0,
  );

  useEffect(() => {
    if (!countriesLoaded) {
      dispatch(fetchCountries());
    }
  }, [dispatch, countriesLoaded]);

  const data = useSelector((state) => state.countries.countries);
  let filteredCountries = filterData(data);
  const [searchQuery, setSearchQuery] = useState('');

  const handleSearchInputChange = (event) => {
    const inputValue = event.target.value;
    const formattedValue = inputValue
      .toLowerCase()
      .replace(/(?:^|\s)\w/g, (match) => match.toUpperCase());
    setSearchQuery(formattedValue);
  };

  filteredCountries = data.filter(
    (country) => country.name.toLowerCase().includes(searchQuery.toLowerCase()),
  );

  return (
    <div>
      <p className="search">
        <input
          type="text"
          placeholder="Search events"
          value={searchQuery}
          onChange={handleSearchInputChange}
        />
      </p>
      <div className="country-container">
        {filteredCountries.map((country, index) => {
          let boxClass = '';

          if (index === 1) {
            boxClass = 'light';
          } else if (index % 4 === 1 || index % 4 === 2) {
            boxClass = 'dark';
          } else {
            boxClass = 'even';
          }
          return (
            <Link to={`/country/${encodeURIComponent(country.name)}`} key={country.name}>
              <div className={`country-box ${boxClass}`} key={country.name}>
                <div className="country-name">{country.name}</div>
                <img
                  className="flag"
                  src={country.flag}
                  alt={`Flag of ${country.name}`}
                />
                <div className="population">
                  Population:
                  {' '}
                  {country.population}
                </div>
              </div>
            </Link>
          );
        })}
      </div>
    </div>
  );
};

export default Country;
