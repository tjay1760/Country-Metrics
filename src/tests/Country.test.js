import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux'; // Import the Provider
import store from '../redux/store';
import Country from '../components/Country';

describe('Country component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Country />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
