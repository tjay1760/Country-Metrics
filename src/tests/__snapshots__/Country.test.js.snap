// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <p
      class="search"
    >
      <input
        placeholder="Search events"
        type="text"
        value=""
      />
    </p>
    <div
      class="country-container"
    />
  </div>
</DocumentFragment>
`;

exports[`CountryDetails component should match snapshot 1`] = `
<DocumentFragment>
  <div>
    ...
  </div>
</DocumentFragment>
`;
